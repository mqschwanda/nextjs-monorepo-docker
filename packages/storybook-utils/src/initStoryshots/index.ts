import type { StoryshotsOptions } from '@storybook/addon-storyshots/dist/api/StoryshotsOptions';
import initAddonStoryshots from '@storybook/addon-storyshots';

/**
 * Options for the `initStoryshots` function.
 */
export type InitStoryshotsOptions = StoryshotsOptions & {
  /**
   * If the snapshot tests fail when react logs a key error.
   *
   * Set this to `false` if stories avoid using the key because it would show up in the code example.
   *
   * @default true
   */
  failOnReactKeyError?: boolean
};

/**
 * A wrapper for the `@storybook/addon-storyshots` module
 *
 * See [addon storyshots docs](https://storybook.js.org/addons/@storybook/addon-storyshots) for more information.
 */
export default function initStoryshots({
  failOnReactKeyError = true,
  ...options
}: InitStoryshotsOptions) {
  if (failOnReactKeyError === false) {
    const consoleError = jest.spyOn(console, 'error');

    beforeAll(() => {
      consoleError.mockImplementation((...args) => {
        const [message] = args;
        if (message.includes('Each child in a list should have a unique "key" prop.')) {
          return;
        }

        console.error(...args); // eslint-disable-line no-console
      });
    });

    afterAll(() => {
      consoleError.mockRestore();
    });
  }

  initAddonStoryshots(options);
}
