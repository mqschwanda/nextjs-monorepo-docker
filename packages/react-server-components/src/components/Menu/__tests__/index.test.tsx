import { render } from '@testing-library/react';
import initStoryshots from '@/__tests__/initStoryshots';
import {
  Menu,
  MenuTitle,
  MenuItem,
} from '@';

initStoryshots({
  storyKindRegex: /^@mqs\/react-server-components\/components\/Menu$/,
});

describe('@mqs/react-server-components', () => {
  describe('components', () => {
    describe('<Menu />', () => {
      it('renders', () => {
        const testId = Menu.name;
        /* eslint-disable jsx-a11y/anchor-is-valid */
        const children = (
          <>
            <MenuTitle>
              <a
                href='#'
              >
                { 'Kittens!' }
              </a>
            </MenuTitle>
            <MenuItem>
              <a
                href='#'
              >
                { 'Normal Kitten' }
              </a>
            </MenuItem>
            <MenuItem
              variantState='active'
            >
              <a
                href='#'
              >
                { 'Active Kitten' }
              </a>
            </MenuItem>
            <MenuItem
              variantState='disabled'
            >
              <a
                href='#'
              >
                { 'Disabled Kitten' }
              </a>
            </MenuItem>
            <MenuItem
              variantState='focus'
            >
              <a
                href='#'
              >
                { 'Focused Kitten' }
              </a>
            </MenuItem>
          </>
        );
        /* eslint-enable jsx-a11y/anchor-is-valid */

        const { getByTestId, asFragment } = render((
          <Menu
            className='test'
            testId={testId}
          >
            { children }
          </Menu>
        ));

        const element = getByTestId(testId);
        expect(element).toBeTruthy();

        expect(asFragment()).toMatchSnapshot();
      });
    });
  });
});
