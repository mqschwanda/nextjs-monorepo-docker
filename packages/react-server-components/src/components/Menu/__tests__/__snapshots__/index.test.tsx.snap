// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@mqs/react-server-components components <Menu /> renders 1`] = `
<DocumentFragment>
  <ul
    class="menu menu-vertical menu-md test"
    data-testid="Menu"
  >
    <li
      class="menu-title"
      data-testid="MenuTitle"
    >
      <a
        href="#"
      >
        Kittens!
      </a>
    </li>
    <li
      class=""
      data-testid="MenuItem"
    >
      <a
        href="#"
      >
        Normal Kitten
      </a>
    </li>
    <li
      class="active"
      data-testid="MenuItem"
    >
      <a
        href="#"
      >
        Active Kitten
      </a>
    </li>
    <li
      class="disabled"
      data-testid="MenuItem"
    >
      <a
        href="#"
      >
        Disabled Kitten
      </a>
    </li>
    <li
      class="focus"
      data-testid="MenuItem"
    >
      <a
        href="#"
      >
        Focused Kitten
      </a>
    </li>
  </ul>
</DocumentFragment>
`;

exports[`Storyshots @mqs/react-server-components/components/Menu Menu Example 1`] = `
<ul
  className="menu menu-vertical menu-md"
  data-testid="Menu"
>
  <li
    className="menu-title"
    data-testid="MenuTitle"
  >
    <a
      href="#"
    >
      Kittens!
    </a>
  </li>
  <li
    className=""
    data-testid="MenuItem"
  >
    <a
      href="#"
    >
      Normal Kitten
    </a>
  </li>
  <li
    className="active"
    data-testid="MenuItem"
  >
    <a
      href="#"
    >
      Active Kitten
    </a>
  </li>
  <li
    className="disabled"
    data-testid="MenuItem"
  >
    <a
      href="#"
    >
      Disabled Kitten
    </a>
  </li>
  <li
    className="focus"
    data-testid="MenuItem"
  >
    <a
      href="#"
    >
      Focused Kitten
    </a>
  </li>
</ul>
`;
